SCClassLibrary/Common/Control/ServerStatus.sc

```diff
@@ -142,20 +142,22 @@ ServerStatusWatcher {
 			} {
 				0.2.wait;
 			};
-			if(serverRunning.not, {
-				if(onFailure.notNil) {
-					postError = (onFailure.value(server) == false);
-				};
-				if(postError) {
-					"Server '%' on failed to start. You may need to kill all servers".format(server.name).error;
-				};
-				serverBooting = false;
-				server.changed(\serverRunning);
-			}, onComplete);
+			if(serverRunning, onComplete, { this.failedToBoot(onFailure) });
```

Comment:
Here, by definition of the `while` exit condition, `serverRunning` must be true. `failedToBoot` cannot ever run.

Reply:
yes, isn't it interesting. This has always been that way, only by refactoring became visible.